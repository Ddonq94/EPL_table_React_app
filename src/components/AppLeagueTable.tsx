import { Button, Table } from "antd";
import { useEffect, useState } from "react";
import { getTableFromJSON } from "../services/dataService";
import AppViewClub from "./AppViewClub";

function AppLeagueTable({ testID = "AppLeagueTable" }: any) {
  const [tableData, setTableData] = useState<any>([]);
  const [visible, setVisible] = useState(false);
  const [currentTeam, setCurrentTeam] = useState<any>("");

  const handleTeamClick = (teamName: any) => {
    setCurrentTeam(teamName);
    setVisible(true);
  };

  useEffect(() => {
    const newTableData = getTableFromJSON();

    setTableData(newTableData);
  }, []);

  const columns: any = [
    {
      title: "Position",
      key: "index",
      render: (text: any, record: any, index: any) => index + 1,
    },
    {
      title: "Club",
      key: "name",
      dataIndex: "name",
      render: (text: any, record: any) => {
        return (
          <Button type="link" onClick={() => handleTeamClick(record.name)}>
            {record.name}
          </Button>
        );
      },
    },
    {
      title: "Played",
      dataIndex: "played",
      key: "played",
    },
    {
      title: "Won",
      dataIndex: "won",
      key: "won",
    },
    {
      title: "Drawn",
      dataIndex: "drawn",
      key: "drawn",
    },
    {
      title: "Lost",
      dataIndex: "lost",
      key: "lost",
    },
    {
      title: "GF",
      dataIndex: "gf",
      key: "gf",
    },
    {
      title: "GA",
      dataIndex: "ga",
      key: "ga",
    },
    {
      title: "GD",
      dataIndex: "gd",
      key: "gd",
      render: (text: any, { gd }: any) => (gd > 0 ? `+${gd}` : gd),
    },
    {
      title: "Points",
      dataIndex: "points",
      key: "points",
    },
  ];

  return (
    <section data-testid={testID} className="AppLeagueTable">
      <h3 className="center-text">Click a club to see fixtures</h3>

      <Table
        columns={columns}
        dataSource={tableData}
        pagination={false}
        rowKey="name"
      />

      {visible && (
        <AppViewClub
          visible={visible}
          currentTeam={currentTeam}
          handleClose={() => setVisible(false)}
        />
      )}
    </section>
  );
}

export default AppLeagueTable;
