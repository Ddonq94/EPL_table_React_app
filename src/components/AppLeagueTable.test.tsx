import React from "react";
import { cleanup, render, screen, fireEvent } from "@testing-library/react";
import AppLeagueTable from "./AppLeagueTable";

afterEach(() => {
  cleanup();
});

describe("League table rendering", () => {
  it("renders AppLeagueTable component", () => {
    const TestFn = () => {
      return <AppLeagueTable testID="AppLeagueTable" />;
    };

    const { getByTestId } = render(<TestFn />);
    const content = getByTestId("AppLeagueTable");
    expect(content).toBeInTheDocument();
    expect(content).toBeDefined();
    expect(content).toBeEnabled();
    expect(
      screen.getByText(/Click a club to see fixtures/i)
    ).toBeInTheDocument();
  });

  it("renders AppLeagueTable component title", () => {
    const TestFn = () => {
      return <AppLeagueTable testID="AppLeagueTable" />;
    };

    render(<TestFn />);

    expect(
      screen.getByText(/Click a club to see fixtures/i)
    ).toBeInTheDocument();
  });

  it("renders AppViewClub on club click from League table", () => {
    const TestFn = () => {
      return <AppLeagueTable testID="AppLeagueTable" />;
    };

    render(<TestFn />);

    fireEvent.click(screen.getByText("Manchester United"));

    expect(
      screen.getByText("Manchester United FC Fixtures")
    ).toBeInTheDocument();
  });

  it("renders AppLeagueTable to match snapshot", () => {
    const TestFn = () => {
      return <AppLeagueTable testID="AppLeagueTable" />;
    };

    const { getByTestId } = render(<TestFn />);
    const content = getByTestId("AppLeagueTable");
    expect(content).toMatchSnapshot();
  });
});
