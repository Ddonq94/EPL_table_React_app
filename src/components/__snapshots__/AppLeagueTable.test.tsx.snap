// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`League table rendering renders AppLeagueTable to match snapshot 1`] = `
<section
  class="AppLeagueTable"
  data-testid="AppLeagueTable"
>
  <h3
    class="center-text"
  >
    Click a club to see fixtures
  </h3>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table"
        >
          <div
            class="ant-table-container"
          >
            <div
              class="ant-table-content"
            >
              <table
                style="table-layout: auto;"
              >
                <colgroup />
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class="ant-table-cell"
                    >
                      Position
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Club
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Played
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Won
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Drawn
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Lost
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      GF
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      GA
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      GD
                    </th>
                    <th
                      class="ant-table-cell"
                    >
                      Points
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="Leicester City"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <div
                          class="club-wrapper  "
                          data-testid="AppClub"
                        >
                          <img
                            alt="Leicester CityLogo"
                            data-testid="AppClubImg"
                            src="https://resources.premierleague.com/premierleague/badges/25/t13.png"
                          />
                          <span>
                             
                            Leicester City
                          </span>
                        </div>
                      </button>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      4
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      8
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      +5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      10
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="Liverpool"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <div
                          class="club-wrapper  "
                          data-testid="AppClub"
                        >
                          <img
                            alt="LiverpoolLogo"
                            data-testid="AppClubImg"
                            src="https://resources.premierleague.com/premierleague/badges/25/t14.png"
                          />
                          <span>
                             
                            Liverpool
                          </span>
                        </div>
                      </button>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      8
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      4
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      +4
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      9
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="Manchester United"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <div
                          class="club-wrapper  "
                          data-testid="AppClub"
                        >
                          <img
                            alt="Manchester UnitedLogo"
                            data-testid="AppClubImg"
                            src="https://resources.premierleague.com/premierleague/badges/25/t1.png"
                          />
                          <span>
                             
                            Manchester United
                          </span>
                        </div>
                      </button>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      11
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      9
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      +2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      5
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="Manchester City"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      4
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <div
                          class="club-wrapper  "
                          data-testid="AppClub"
                        >
                          <img
                            alt="Manchester CityLogo"
                            data-testid="AppClubImg"
                            src="https://resources.premierleague.com/premierleague/badges/25/t43.png"
                          />
                          <span>
                             
                            Manchester City
                          </span>
                        </div>
                      </button>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      4
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      4
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      -1
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      4
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="Tottenham Hotspur"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <div
                          class="club-wrapper  "
                          data-testid="AppClub"
                        >
                          <img
                            alt="Tottenham HotspurLogo"
                            data-testid="AppClubImg"
                            src="https://resources.premierleague.com/premierleague/badges/25/t6.png"
                          />
                          <span>
                             
                            Tottenham Hotspur
                          </span>
                        </div>
                      </button>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="Chelsea"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      6
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <div
                          class="club-wrapper  "
                          data-testid="AppClub"
                        >
                          <img
                            alt="ChelseaLogo"
                            data-testid="AppClubImg"
                            src="https://resources.premierleague.com/premierleague/badges/25/t8.png"
                          />
                          <span>
                             
                            Chelsea
                          </span>
                        </div>
                      </button>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      10
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      -5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      3
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row ant-table-row-level-0"
                    data-row-key="Arsenal"
                  >
                    <td
                      class="ant-table-cell"
                    >
                      7
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      <button
                        class="ant-btn ant-btn-link"
                        type="button"
                      >
                        <div
                          class="club-wrapper  "
                          data-testid="AppClub"
                        >
                          <img
                            alt="ArsenalLogo"
                            data-testid="AppClubImg"
                            src="https://resources.premierleague.com/premierleague/badges/25/t3.png"
                          />
                          <span>
                             
                            Arsenal
                          </span>
                        </div>
                      </button>
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      4
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      0
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      10
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      -5
                    </td>
                    <td
                      class="ant-table-cell"
                    >
                      2
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
